Compilation issue with -O0 flag

I am running into the following error when  compiling bagel without optimization (`-O0` or `-O1`), using gcc 9.4

```cpp
In file included from ../../../../src/asd/dmrg/dmrg_block.h:30,
                 from ../../../../src/asd/dmrg/product_civec.h:31,
                 from ../../../../src/ci/ras/apply_operator.h:30,
                 from ../../../../src/ci/ras/apply_operator.cc:26:
../../../../src/asd/dmrg/gamma_forest_asd.h: In constructor ‘bagel::GammaForestASD<VecType>::GammaForestASD(std::map<bagel::BlockKey, std::shared_ptr<const _Tp> >)’:
../../../../src/asd/dmrg/gamma_forest_asd.h:81:64: error: invalid operands of types ‘const char [4]’ and ‘bool’ to binary ‘operator<<’
   81 |                 std::cout << is_alpha(*opiter) ? "(A)" : "(B)" << is_creation(*opiter) ? "^t" : "";
      |                                                          ~~~~~ ^~ ~~~~~~~~~~~~~~~~~~~~
      |                                                          |                   |
      |                                                          const char [4]      bool

```

The following patch fixes this:

```diff
diff --git a/src/asd/dmrg/gamma_forest_asd.h b/src/asd/dmrg/gamma_forest_asd.h
index c04e0695e..c8781e550 100644
--- a/src/asd/dmrg/gamma_forest_asd.h
+++ b/src/asd/dmrg/gamma_forest_asd.h
@@ -78,7 +78,7 @@ class GammaForestASD : public GammaForest<VecType, 1> {
 #ifdef DEBUG
               std::cout << "inserting: <" << bra.first.nelea << ", " << bra.first.neleb << "|";
               for (auto opiter = coupling.begin(); opiter != coupling.end(); ++opiter)
-                std::cout << is_alpha(*opiter) ? "(A)" : "(B)" << is_creation(*opiter) ? "^t" : "";
+                std::cout << (is_alpha(*opiter) ? "(A)" : "(B)") << (is_creation(*opiter) ? "^t" : "");
               std::cout << "|" << ket.first.nelea << ", " << ket.first.neleb << ">" << std::endl;
 #endif
               sparselist_.emplace_back(coupling, BlockInfo(bra.first.nelea, bra.first.neleb, bra.second->ij()), BlockInfo(ket.first.nelea, ket.first.neleb, ket.second->ij()));
```

Happy to submit a PR and sign a copyright.

--- a/src/asd/dmrg/gamma_forest_asd.h
+++ b/src/asd/dmrg/gamma_forest_asd.h
@@ -78,7 +78,7 @@ class GammaForestASD : public GammaForest<VecType, 1> {
 #ifdef DEBUG
               std::cout << "inserting: <" << bra.first.nelea << ", " << bra.first.neleb << "|";
               for (auto opiter = coupling.begin(); opiter != coupling.end(); ++opiter)
-                std::cout << is_alpha(*opiter) ? "(A)" : "(B)" << is_creation(*opiter) ? "^t" : "";
+                std::cout << (is_alpha(*opiter) ? "(A)" : "(B)") << (is_creation(*opiter) ? "^t" : "");
               std::cout << "|" << ket.first.nelea << ", " << ket.first.neleb << ">" << std::endl;
 #endif
               sparselist_.emplace_back(coupling, BlockInfo(bra.first.nelea, bra.first.neleb, bra.second->ij()), BlockInfo(ket.first.nelea, ket.first.neleb, ket.second->ij()));

